platform/o.n.swing.laf.flatlaf/src/org/netbeans/swing/laf/flatlaf/ui/FlatViewTabDisplayerUI.java

```diff
@@ -232,7 +232,9 @@ private void paintTabBackgroundAtScale1x(Graphics2D g, int index, int width, int
             g.fillRect(0, height - contentBorderWidth, width, contentBorderWidth);
         }
 
-        if (showTabSeparators && index >= 0) {
+        if (showTabSeparators && index >= 0 &&
+                ((!isSelected(index) && !isSelected(index + 1)
+                || selectedBackground.equals(activeBackground)))) {
```

Comment:
Good change. Do you need to check index < getDataModel().size() before isSelected(index + 1)?

Reply:
Underlying code supports out of range values.